### PR TITLE
fix(ext/websockets): ensure we fully send frames before close

### LINKED ISSUE #19483: 1.34.2  Websocket Regressiong: Closeframe send before flushed
Works in 1.34.1:

```ts
import { serve } from "https://deno.land/std@0.191.0/http/mod.ts";
serve((req) => {
    const { response, socket } = Deno.upgradeWebSocket(req);
    socket.onopen = () => socket.send("Hi");
    socket.onmessage = () => {
        socket.send("Bye");
        socket.close();
    };
    return response;
}, { signal: AbortSignal.timeout(100) });
const ws = new WebSocket("ws://localhost:8000");
let hasSeen = false;
ws.onmessage = ({ data }) => {
    if (data == "Hi")
        ws.send("Hey!");
    if (data == "Bye")
        hasSeen = true;
};
ws.onclose = () => {
    if (!hasSeen)
        throw new Error("Not Found");
    else console.log("found");
};
```

### PR DESCRIPTION
**Pull request by mmastrac**
Fixes #19483

### CODE REVIEW

**Review by aapoalas — APPROVED** [review]
Looks good to me.

File: cli/tests/unit/websocket_test.ts [hunk 1]
```diff
@@ -100,7 +100,7 @@ Deno.test(
       promise.resolve();
     };
 
-    await Promise.all([promise, server]);
+    await Promise.all([promise, server.finished]);
```
[the remark this answers is not in the document]
Comment (mmastrac) [reply cli/tests/unit/websocket_test.ts:103]: @aapoalas I would love to figure out how to automate finding these!

[the remark this answers is not in the document]
Comment (aapoalas) [reply cli/tests/unit/websocket_test.ts:103]: Yeah. I guess a lint would be the right choice.

[the remark this answers is not in the document]
Comment (mmastrac) [reply cli/tests/unit/websocket_test.ts:103]: Oh huh... I think I found a way. PR incoming...

[the remark this answers is not in the document]
Comment (aapoalas) [reply cli/tests/unit/websocket_test.ts:103]: Oooh!

**Pull request merged**